### PR TITLE
feat(radio): add new properties and prepare for development

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Developers must use `ion-item` to add a label to `ion-radio`. 

Issue URL: Internal

## What is the new behavior?

- `labelPlacement` and `justify` properties are available on `ion-radio`
- `ion-radio` has a default slot for rendering labels
- `ion-radio` can render either the legacy or modern template
  - Developers will receive a deprecation warning for legacy radio templates
- Moves legacy tests to `radio/test/legacy` directory

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The modern basic tests will be added/template updated in future PRs.

### CODE REVIEW

File: core/src/components/radio/radio.tsx [hunk 1]
```diff
@@ -143,6 +172,26 @@ export class Radio implements ComponentInterface {
...
+
+  private renderRadio() {
+    return <slot></slot>;
+  }
+
+  private renderLegacyRadio() {
+    if (!this.hasLoggedDeprecationWarning) {
+      printIonWarning(
+        `Using ion-radio with an ion-label has been deprecated. To migrate, remove the ion-label and pass your label directly into ion-radio instead.
+Example: <ion-radio>Email:</ion-radio>
```
Comment (sean-perkins) [inline core/src/components/radio/radio.tsx:188]: I remained consistent with toggle here, but I am questioning if showing a code snippet like:
```html
<ion-radio>Label:</ion-radio>
```
May be more intuitive/clear.

Comment (liamdebeasi) [reply core/src/components/radio/radio.tsx:188]: Not sure I understand -- the code snippets look the same? (minus the label text)

**Review by amandaejohnston — APPROVED** [review]

File: core/src/components/radio/radio.tsx (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/radio/radio.tsx:188]: Correct, I think the default label text is a little confusing. Why would I have a radio with an `Email` label? `Option Label` would also work for me.

Comment (liamdebeasi) [reply core/src/components/radio/radio.tsx:188]: Oh I see now 🤦 That's likely a copy and paste mistake from the ion-input revision.

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**